templateflow/conf/_s3.py

```diff
@@ -80,23 +80,32 @@ def _update_skeleton(skel_file, dest, overwrite=True, silent=False):
     dest = Path(dest)
     dest.mkdir(exist_ok=True, parents=True)
     with ZipFile(skel_file, 'r') as zipref:
+        allfiles = sorted(zipref.namelist())
+
         if overwrite:
-            zipref.extractall(str(dest))
-            return True
+            newfiles = allfiles
+        else:
+            current_files = [s.relative_to(dest) for s in dest.glob('**/*')]
+            existing = sorted({'%s/' % s.parent for s in current_files}) + [
+                str(s) for s in current_files
+            ]
+            newfiles = sorted(set(allfiles) - set(existing))
 
-        allfiles = zipref.namelist()
-        current_files = [s.relative_to(dest) for s in dest.glob('**/*')]
-        existing = sorted({'%s/' % s.parent for s in current_files}) + [
-            str(s) for s in current_files
-        ]
-        newfiles = sorted(set(allfiles) - set(existing))
         if newfiles:
             if not silent:
                 print(
                     'Updating TEMPLATEFLOW_HOME using S3. Adding:\n%s'
                     % '\n'.join(newfiles)
                 )
-            zipref.extractall(str(dest), members=newfiles)
+            for fl in newfiles:
+                localpath = dest / fl
+                if localpath.exists():
+                    continue
+                try:
+                    zipref.extract(fl, path=dest)
+                except FileExistsError:
+                    # If there is a conflict, do not clobber
```

Comment:
Increasingly looking like we'll need to check hashes at some point. If a template gets updated this will leave the old file even with `overwrite=True`, if I understand properly.

Reply:
Yes, some form of versioning for individual template files will be needed for us to determine if an "old" file needs to be updated, but I don't think that falls in the scope of this PR. Now that I think of it, I don't fully understand `overwrite`'s intention - remove all cached templates and force them to be pulled again?

Reply:
Re: scope of this PR - I totally agree, it should've been more clear in my message.

Re: `overwrite` intention - I don't know it either (meaning, happy to review suggestions in the future clarifying/changing it).